Update game.js for sonar pulse mode hover feature

kramerje opened the pull request:
implemented the hover feature for sonar pulse mode. I forgot to comment out the debugging mode (see line 188 @thenellg).

aalokb commented on src/main/java/assets/game.js:
```diff
@@ -185,11 +185,11 @@ function redrawGrid() {
-////        document.getElementById("opponent").rows[square.row-1].cells[square.column.charCodeAt(0) - 'A'.charCodeAt(0)].classList.add("occupied");
-//    }));
+
+        document.getElementById("opponent").rows[square.row-1].cells[square.column.charCodeAt(0) - 'A'.charCodeAt(0)].classList.add("occupied");
```
Glad this finally worked out, so now we have this dot for only the most recent sonar call. Nice.

aalokb commented on src/main/java/assets/game.js:
```diff
@@ -474,10 +473,63 @@ function hideInstructions(){
+            for (j=0; j<5; j++){
+                // have if statements for the corners to not include on hover
+                // top row
+                if (i === 0 && j === 0 || i === 1 && j === 0 || i === 3 && j === 0 || i === 4 && j === 0){
```
Add parenthesis here, I dont understand the logic of how this conditional is executed... Please in the future make it more clear

kramerje replied:
Sorry, I thought my comments were sufficient. I'll be more thorough in the future. Also, I'll add in parenthesis in the future as well.

aalokb commented on src/main/java/assets/game.js:
```diff
@@ -474,10 +473,63 @@ function hideInstructions(){
+                let cell;
+
+                // check for when cursor is at edge of board
+                if (table.rows[row - 2 + j] === undefined){
```
The continue in all of these statements is not neccessary..

kramerje replied:
Without them, the hover errors because it tries to access undefined. Basically, when the cursor is near the edge of the grid, the full hover sonar range isn't extended fully because of this. I believe they are necessary.

kramerje commented:
@NikhilAnand1998: I changed it to board so that function can handle more than one board. Before the hover function, it just needed to look at players board since the register listener was only used for placing ships. Now, we want something to hover on the opponent's board. So I changed it to be able to handle that.
